### PR TITLE
docs: update versioning documentation

### PR DESCRIPTION
**Pull request by gldgrnt**

### CODE REVIEW

File: doc/contributing/versioning.md [hunk 1]
```diff
@@ -1,131 +1,218 @@
...
+### 1.2.3 New branches
 
-Make an active effort to involve the community, this might be in the form of presentations or meetings.
+In order to add to the Design System, you must first create a new branch with the following format `xx/type/short-description`.
 
-## Deprecation
+Where:
 
-Deprecation is the practice of communicating features in the library that should no longer be used and requires a user change behaviour in their application.
+- `xx` is your first and last intials, e.g. `jd` for John Doe
```
Comment (catiarodriguescosta) [inline doc/contributing/versioning.md:61]:
```suggestion
- `xx` is your first and last initials, e.g. `jd` for John Doe
```

File: doc/contributing/versioning.md [hunk 2]
```diff
@@ -1,131 +1,218 @@
...
-- [JavaScript](https://github.com/alphagov/wmnds-frontend/blob/master/docs/installation/installing-with-npm.md#using-javascript)
-- SCSS - https://wmnds-frontend-review.herokuapp.com/docs/
-- Nunjucks Macros (Templates)
+- `SEMVER_GITHUB_TOKEN` - a personal access token with repo permissions. See the [semantic-release configuration](https://github.com/semantic-release/github#configuration) for details.
+- `NETLIFY_AUTH_TOKEN` - Netlify personal access token. See [actions-netlify](https://github.com/nwtgck/actions-netlify#required-inputs-and-env) for details.
+- `NETLIFY_SITE_ID` - Netlify API ID. See [actions-netlify](https://github.com/nwtgck/actions-netlify#required-inputs-and-env) for details.
+- `NPM_TOKEN` - npm access token with publish permissions. See the [semantic-release configuration](https://github.com/semantic-release/github#configuration) for details.
 
-## Updating Changelog
+The workflow also requires a `GITHUB_TOKEN` but that is made avaible automatically during any workflow run.
```
Comment (catiarodriguescosta) [inline doc/contributing/versioning.md:116]:
```suggestion
The workflow also requires a `GITHUB_TOKEN` but that is made available automatically during any workflow run.
```

**Review by catiarodriguescosta — APPROVED** [review]
Looks pretty good and complete! 👍 
Just flag a few typos.

**Review by daylesalmon — APPROVED** [review]

**Pull request merged**